src/__integrations__/uat-prod/pmClient.resources.integration.uat-prod.test.ts

```diff
@@ -0,0 +1,117 @@
+// import { fromNullable } from 'fp-ts/lib/Option';
+import { Millisecond } from 'italia-ts-commons/lib/units';
+import 'abort-controller/polyfill';
+import nodeFetch from 'node-fetch';
+import { createClient } from '../../../generated/definitions/pagopa/client';
+import { retryingFetch } from '../../utils/fetch';
+import * as myFetch from '../../utils/fetch';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any,functional/immutable-data
+(global as any).fetch = nodeFetch;
+
+describe('Endpoint resources test suite for UAT/PROD PM', () => {
+  // Set the testing environment
+
+  const PM_PORT = process.env.PAYMENT_MANAGER_PORT ? parseInt(process.env.PAYMENT_MANAGER_PORT, 10) : 1234;
+  const PM_HOST_PROD = process.env.PAYMENT_MANAGER_HOST_PROD as string;
+  const PM_HOST_UAT = process.env.PAYMENT_MANAGER_HOST_UAT as string;
+
+  // eslint-disable-next-line functional/no-let
+  let pmClientUAT;
+  // eslint-disable-next-line functional/no-let
+  let pmClientPROD;
+
+  beforeAll(async () => {
+    // Start clients
+
+    pmClientUAT = createClient({
+      baseUrl: `https://${PM_HOST_UAT}:${PM_PORT}`,
+      fetchApi: retryingFetch(fetch, 5000 as Millisecond, 5),
+    });
+
+    pmClientPROD = createClient({
+      baseUrl: `https://${PM_HOST_PROD}:${PM_PORT}`,
+      fetchApi: retryingFetch(fetch, 5000 as Millisecond, 5),
+    });
+  });
+
+  beforeEach(async () => {
+    const mySpyCustomFetch = jest.spyOn(myFetch, 'retryingFetch');
+
+    const paymentManagerClientUAT = createClient({
+      baseUrl: `https://${PM_HOST_UAT}:${PM_PORT}/pp-restapi` as string,
+      fetchApi: retryingFetch(fetch, 5000 as Millisecond, 5),
+    });
+    const paymentManagerClientPROD = createClient({
+      baseUrl: `https://${PM_HOST_PROD}:${PM_PORT}/pp-restapi` as string,
+      fetchApi: retryingFetch(fetch, 5000 as Millisecond, 5),
+    });
+    // Spy on th global fetch to check if it gets called
+    const mySpyGlobalFetch = jest.spyOn(global, 'fetch');
+    expect(paymentManagerClientUAT).toBeTruthy();
```

Comment:
Why is this test done in _beforeEach_? What is the goal?

Reply:
Tnx for pointing out it, it isn't useful, it is the result of a copy and paste from another test. I'll remove it.